use new google embedded API for youtube

hrnr opened the pull request:
this allows to use HTML5 instead of flash.

Citing google:

> Deprecation Notice
> YouTube `<object>` embeds were deprecated on January 27, 2015. Please migrate your applications to use `<iframe>` embeds, which can intelligently use whichever embedded player – HTML (`<video>`) or Flash (`<object>`) – the client supports.

and

> Best practice: IFrame embeds are the recommended method for embedding a YouTube player because the IFrame will select the appropriate player based on the client's capabilities and available YouTube file formats.

I have added `<center>` to follow the same formatting as Vimeo videos. Minimal size is based on google recomendation.

> Note: Embedded players must have a viewport that is at least 200px by 200px. If the player displays controls, it must be large enough to fully display the controls without shrinking the viewport below the minimum size. We recommend 16:9 players be at least 480 pixels wide and 270 pixels tall.

Reference:
https://developers.google.com/youtube/player_parameters

EDIT: fix style

wjwwood commented:
@hrnr thanks for the patch! I've tested this out using our Vagrant setup for the wiki (though I had to make a patch to the configuration of the wiki by ssh'ing into the machine and change the access control).

The only thing that isn't working is the suppression of related videos (the third item) described here:

http://wiki.ros.org/WikiMacros#Youtube

Do you think you could adapt this PR to support that too?

hrnr commented:
I have fixed that. Also I have noticed that fullscreen is blocked in HTML5 version so I have fixed that too.

Fix for the `&rel=0` is quite nasty, see the commit message. Basically the tip should be `?rel=0`. But I don't think it's worth to break compatibility. (If breaking compat, shouldn't the `&rel=0` be the default?)

wjwwood commented:
@hrnr Yeah... The fix looks ok to me. If you think the `&rel=0` is wrong and `?rel=0` works as well, feel free to update: http://wiki.ros.org/WikiMacros#Youtube

I'll test this again real quick before merging and deploying. Thanks!

wjwwood commented:
lgtm

wjwwood commented:
Looks like it is properly deployed, based on the example page: http://wiki.ros.org/WikiMacros#Youtube

hrnr commented:
ups, videos works fine, but whole page is missing after first youtube video (everywhere). I'm not sure of cause, in html all looks good, and video works fine. Maybe something in generating?

wjwwood commented:
Yeah... didn't notice that. I'm looking into it.

wjwwood commented:
I reverted the deployment on the live site for now. Do you have time to look into the cause? It's not immediately obvious to me why this happening. The tags seem to be closing correctly.

wjwwood commented:
I also didn't see any errors in the apache logs (which normally happens when an exception is raised or something like that).

dirk-thomas commented on macro/Youtube.py:
```diff
@@ -4,4 +4,5 @@ def execute(macro, args):
     key = key[len('http://www.youtube.com/watch?v='):]
-  return macro.formatter.rawHTML('<object width="480" height="295"><param name="movie" value="http://www.youtube.com/v/%(key)s&hl=en_US&fs=1&hd=1"></param><param name="allowFullScreen" value="true"></param><param name="allowscriptaccess" value="always"></param><embed src="http://www.youtube.com/v/%(key)s&hl=en_US&fs=1&hd=1" type="application/x-shockwave-flash" allowscriptaccess="always" allowfull ...
+  key = key.replace('&', '?', 1) # query params must start with ?
+  return macro.formatter.rawHTML('<center><iframe type="text/html" width="480" height="270" src="http://www.youtube.com/embed/%(key)s" frameborder="0" allowfullscreen /></center>'%locals())
```
Maybe try: `allowfullscreen="true"`

wjwwood replied:
I tried that and it still cuts the page off after the Youtube entry. It's like the server stops sending html after the `</center>`.

hrnr commented:
I think I have figured it out problem was with the closing `/>`. Also I have noticed some inconsistencies between youtube and vimeo videos. Created #161, #162.

wjwwood commented:
Great! I merged and deployed both. They seem to be working in my local testing as well as on the live site (the right size, using html5, allows fullscreen and doesn't cut off content).

Thanks for looking into it.

hrnr commented:
thanks for quick deployment. All looks good to me too on the wiki.
